doc/source/guide/in_depth/functional_guide.rst

```diff
@@ -0,0 +1,189 @@
+| ``(S + T)(x)``     | ``S(x) + T(x)`` | `FunctionalSum`                                                                |
+|                    |                 | - Retains `Functional.gradient`.                                               |
++--------------------+-----------------+--------------------------------------------------------------------------------+
+| ``(S + a)(x)``     | ``S(x) + a``    | `FunctionalScalarSum`                                                          |
+|                    |                 | - Retains all properties.                                                      |
+|                    |                 | Note that this never means scaling of the argument.                            |
++--------------------+-----------------+--------------------------------------------------------------------------------+
+| ``(S * A)(x)``     | ``S(A(x))``     | `FunctionalComp`                                                               |
+|                    |                 | - Retains `Functional.gradient`.                                               |
++--------------------+-----------------+--------------------------------------------------------------------------------+
+| ``(a * S)(x)``     | ``a * S(x)``    | `FunctionalLeftScalarMult`                                                     |
+|                    |                 | - Retains all properties if ``a`` is positive.                                 |
+|                    |                 | Otherwise only `Functional.gradient` and `Functional.derivative` are retained. |
++--------------------+-----------------+--------------------------------------------------------------------------------+
+| ``(S * a)(x)``     | ``S(a * x)``    | `FunctionalRightScalarMult`                                                    |
+|                    |                 | - Retains all properties.                                                      |
++--------------------+-----------------+--------------------------------------------------------------------------------+
+| ``(v * S)(x)``     | ``v * S(x)``    | `FunctionalLeftVectorMult`                                                     |
+|                    |                 | - Results in an operator rather than a functional.                             |
++--------------------+-----------------+--------------------------------------------------------------------------------+
+| ``(S * v)(x)``     | ``S(v * x)``    | `FunctionalRightVectorMult`                                                    |
+|                    |                 | - Retains gradient and convex conjugate.                                       |
++--------------------+-----------------+--------------------------------------------------------------------------------+
+| ``f.translate(y)`` | ``f(. - y)``    | `TranslatedFunctional`                                                         |
```

Comment:
Should probably be called `FunctionalTranslated` to stay within the naming bounds.